src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -80,6 +84,11 @@ public void update(String caPassphrase, CertificateStore.CAType caType) throws K
         certificateStore.update(caPassphrase, caType);
     }
 
+    public void setCA(PrivateKey privateKey, X509Certificate... certificates) {
+        certificateStore.setCaCertificateChain(certificates);
+        certificateStore.setCaPrivateKey(privateKey);
```

Comment:
This doesn't seem like the correct interface. You shouldn't be able to update these independently or else you can end up in an inconsistent state

Reply:
Updated.